state/accumulator.go

```diff
@@ -317,6 +317,16 @@ func (a *Accumulator) Initialise(roomID string, state []json.RawMessage) (Initia
 	return res, err
 }
 
+type AccumulateResult struct {
+	// TODO: is this redundant---identical to len(TimelineNIDs)?
```

Comment:
I don't think so. If you call accumulate 2x with:
 - a,b,c
 - a,b,c,d

Then in the latter case you get timeline nids x4 and num new = 1.

Reply:
Hmm. It's a bit surprising to me that we embed all of these events---including those which we know are _not_ new---in the V2Accumulate payload. Is the point that the pubsub might arrive out-of-order though? (E.g. Seeing (d) then (a, b, c) would be bad.)

Reply:
> Then in the latter case you get timeline nids x4 and num new = 1.

I think this is wrong.

- The second Accumulate call would return deduped events `[d]` thanks to this logic: https://github.com/matrix-org/sliding-sync/blob/6f3f556842a5e04c3ce67aec4204292f3e55c9c3/state/accumulator.go#L598 
- numNew should be the same as len(dedupedEvents) due to this: https://github.com/matrix-org/sliding-sync/blob/main/state/accumulator.go#L396-L404
- timlineNIDs corresponds to dedupedEvents thanks to this loop: https://github.com/matrix-org/sliding-sync/blob/main/state/accumulator.go#L409-L438